### DISCUSSION TITLE
Mobx Upgrade With Base Classes (v4/5 to 6)

### DISCUSSION THREAD

**Discussion by LogicRoom** [mobxjs/mobx#2691] [Q&A]
We have some classes....

```
export class CustomField extends GenericField {
   @observable customProperty
}
```

```
class GenericField {
   @observable genericProperty
}
```

We ran the codemod to upgrade Mobx as per the documentation at..... https://mobx.js.org/migrating-from-4-or-5.html

The codemod simply added

```
constructor() {
   makeObservable(this);
}
```

To the base class (not the child class) and came up with the error:

**/src/CustomField.ts Transformation error (Cannot read property 'start' of null) TypeError: Cannot read property 'start' of null**

We are unsure of what to do next? The documentation is not clear in this regard https://mobx.js.org/migrating-from-4-or-5.html#limitations-of-mobx-undecorate it says if base classes expect arguments (does this mean through the constructor?) the codemod will not work. But it doesn't expect anything, so this statement doesn't make sense and nor does the warning.

If we run the codemod again it adds yet another call to makeObservable in the base class but nothing changes on the child.

Any ideas what the correct way forward is....?

**Comment by mweststrate** [comment]
Every class that introduces observables needs makeObservable, so that
includes the subclass indeed.

Op ma 28 dec. 2020 09:30 schreef Logic Room <notifications@github.com>:

**Comment by LogicRoom** [comment]
Thanks @mweststrate , in that case I think the codemod has a bug? 

The documentation says ....

**The mobx-undecorate command has to introduce a constructor in classes that do not yet have one. If base class of the constructor expects arguments, the codemod cannot introduce these arguments for the subclass being upgraded, and the super call won't pass them either. You have to fix these manually**

Yet our class (GenericField ) does not have a constructor argument in the base class, so in that case there is no reason the codemod cannot create one in the subclass? 

Also the warning says... 'Transformation error (Cannot read property 'start' of null)' which isn't clear what the problem is?

**Comment by mweststrate** [comment]
The codemod is a best effort, not a guaranteed correct / complete
conversion, and many things can influence its results, for example certain
Babel syntax extensions might make the file unparsable for the codemod. If
you think a case is missing that should be supported, please PR a unit test
with a reproduction.

Op ma 28 dec. 2020 11:14 schreef Logic Room <notifications@github.com>:

**Reply by LogicRoom** [reply]
Hi @mweststrate we are doing a significant upgrade with a client and also use Mobx in our UI architecture training. As soon as we finish this upgrade we will look at putting a PR in for this particular issue. Thanks!
